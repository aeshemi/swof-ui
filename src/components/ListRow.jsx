import React from 'react';
import { Media, Panel, Row, Grid } from 'react-bootstrap';
import Shift from './Shift';

const ListRow = props => {
  const { day, dayOfWeek, engineer } = props;

  return (
    <Media>
      <Media.Left align="top">
        <Panel bsStyle="primary" className="date">
          <Panel.Heading className="day-of-week">{dayOfWeek}</Panel.Heading>
          <Panel.Body className="day">{day}</Panel.Body>
        </Panel>
      </Media.Left>
      <Media.Body>
        <Grid>
          <Row>
            {Object.keys(engineer).map(key => (
              <Shift key={key} shift={key} name={engineer[key]} />
            ))}
          </Row>
        </Grid>
      </Media.Body>
    </Media>
  );
};

export default ListRow;
