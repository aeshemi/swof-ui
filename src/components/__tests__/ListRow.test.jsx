import ListRow from '../ListRow';

it('ListRow renders correctly', () => {
  const wrapper = shallow(
    <ListRow
      key={19}
      day={19}
      dayOfWeek="Wed"
      engineer={{ Morning: 'Nap Nimmo', Afternoon: 'Monti Vercruysse' }}
    />
  );

  expect(wrapper).toMatchSnapshot();
});
