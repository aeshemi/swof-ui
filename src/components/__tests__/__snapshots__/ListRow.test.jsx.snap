// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListRow renders correctly 1`] = `
<Media
  bsClass="media"
  componentClass="div"
>
  <MediaLeft
    align="top"
    bsClass="media-left"
  >
    <Uncontrolled(Panel)
      bsStyle="primary"
      className="date"
    >
      <PanelHeading
        bsClass="panel"
        className="day-of-week"
        componentClass="div"
      >
        Wed
      </PanelHeading>
      <PanelBody
        bsClass="panel"
        className="day"
        collapsible={false}
      >
        19
      </PanelBody>
    </Uncontrolled(Panel)>
  </MediaLeft>
  <MediaBody
    bsClass="media-body"
    componentClass="div"
  >
    <Grid
      bsClass="container"
      componentClass="div"
      fluid={false}
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Shift
          key="Morning"
          name="Nap Nimmo"
          shift="Morning"
        />
        <Shift
          key="Afternoon"
          name="Monti Vercruysse"
          shift="Afternoon"
        />
      </Row>
    </Grid>
  </MediaBody>
</Media>
`;
